src/OpenTK.Mathematics/Vector/Vector4h.cs

```diff
@@ -120,49 +120,55 @@ public Vector4h(float x, float y, float z, float w)
         /// <summary>
         /// Initializes a new instance of the <see cref="Vector4h"/> struct.
         /// </summary>
-        /// <param name="v">The <see cref="Vector4"/> to convert.</param>
-        public Vector4h(Vector4 v)
+        /// <param name="xy">The x and y components of the Vector4h.</param>
+        /// <param name="z">The z component of the Vector4h.</param>
+        /// <param name="w">The w component of the Vector4h.</param>
+        public Vector4h(Vector2h xy, float z = default, float w = default)
         {
-            X = (Half)v.X;
-            Y = (Half)v.Y;
-            Z = (Half)v.Z;
-            W = (Half)v.W;
+            X = xy.X;
+            Y = xy.Y;
+            Z = (Half)z;
+            W = (Half)w;
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Vector4h"/> struct.
         /// </summary>
-        /// <param name="v">The <see cref="Vector4"/> to convert.</param>
-        public Vector4h(in Vector4 v)
+        /// <param name="xy">The x and y components of the Vector4h.</param>
+        /// <param name="z">The z component of the Vector4h.</param>
+        /// <param name="w">The w component of the Vector4h.</param>
+        public Vector4h(Vector2h xy, Half z = default, Half w = default)
         {
-            X = (Half)v.X;
-            Y = (Half)v.Y;
-            Z = (Half)v.Z;
-            W = (Half)v.W;
+            X = xy.X;
+            Y = xy.Y;
+            Z = z;
+            W = w;
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="Vector4h"/> struct.
         /// </summary>
-        /// <param name="v">The <see cref="Vector4d"/> to convert.</param>
-        public Vector4h(Vector4d v)
+        /// <param name="xyz">The x, y and z components of the Vector4h.</param>
+        /// <param name="w">The w component of the Vector4h.</param>
+        public Vector4h(Vector3h xyz, float w = default)
```

Comment:
We want another version of this ctor that takes `Vector3, float`.

Reply:
See https://github.com/opentk/opentk/pull/1419#discussion_r845903852